### PR TITLE
Require Maven 3.5.0 as minimum, update test dependencies

### PR DESCRIPTION
**Pull request by joerghoh**

### CODE REVIEW

**Comment by davidjgonzalez** [comment]
@joerghoh have you validated this doesn't break anything w/ the release process? In the past, these POM adjustments have messed up the release cause it to take hours to figure out what's wrong/resolve.
Just want to make sure this is properly tested to we dont run into this again.
/cc @badvision

**Comment by joerghoh** [comment]
@davidjgonzalez well, if the build with Travis runs successfully, why do you expect the release process to break? I was never able to reproduce your release issue.

File: pom.xml [hunk 1]
```diff
@@ -103,7 +103,7 @@
                 <plugin>
                     <groupId>org.apache.sling</groupId>
                     <artifactId>maven-sling-plugin</artifactId>
-                    <version>2.3.8</version>
+                    <version>2.4.2</version>
```
Comment (kwin) [inline pom.xml:106]: let us migrate to the latest version with a different GAV: sling-maven-plugin

**Comment by joerghoh** [comment]
I added the milestone 5.1.0 to change these build details.

**Comment by kwin** [comment]
Superseded by #2700

**Pull request closed without merging**